website/docs/reference/global-configs/legacy-behaviors.md

```diff
@@ -12,16 +12,77 @@ Another category of flags provides existing projects with a migration window for
-1. **Introduction:** The flag is introduced, and the logic added within dbt to support both 'old' + 'new' behaviours.
-2. **Maturity:** The default value of the flag is switched, from 'old' to 'new.'
-3. **Deprecation:** The flag, having been marked for deprecation (with user warnings), is removed. The logic to support the ‘old’ behaviour is removed from dbt codebases.
+1. **Introduction (disabled by default):** dbt adds logic to support both 'old' + 'new' behaviors. The 'new' behavior is gated behind a flag, disabled by default, preserving the old behavior.
+2. **Maturity (enabled by default):** The default value of the flag is switched, from `false` to `true`, enabling the new behavior by default. Users can preserve the 'old' behavior and opt out of the 'new' behavior by setting the flag to `false` in their projects. They may see deprecation warnings when they do so.
+3. **Removal (generally enabled):** After marking the flag for deprecation, we remove it along with the 'old' behavior it supported from the dbt codebases. We aim to support most flags indefinitely, but we're not committed to supporting them forever. If we choose to remove a flag, we'll offer significant advance notice.
 
-These flags can _only_ be set in `dbt_project.yml`. They configure behaviors closely tied to project code. They should be defined in version control, and changed via pull/merge request, with the same testing and peer review.
+## Behavior change flags
 
-## Behaviors
+These flags _must_ be set in the `flags` dictionary in `dbt_project.yml`. They configure behaviors closely tied to project code, which means they should be defined in version control and modified through pull or merge requests, with the same testing and peer review.
+
+The following example displays the current flags and their current default values in the latest dbt Cloud and dbt Core versions. To opt out of a specific behavior change, set the values of the flag to `False` in `dbt_project.yml`. You'll continue to see warnings for legacy behaviors that you have opted out of explicitly until you either resolve them (switch the flag to `True`) or choose to silence the warnings.
+
+<File name='dbt_project.yml'>
+
+```yml
+flags:
+  require_explicit_package_overrides_for_builtin_materializations: False
+  require_model_names_without_spaces: False
+  source_freshness_run_project_hooks: False
+```
+
+</File>
+
+| Flag                                                            | dbt Cloud: Intro | dbt Cloud: Maturity | dbt Core: Intro | dbt Core: Maturity | 
+|-----------------------------------------------------------------|------------------|---------------------|-----------------|--------------------|
+| require_explicit_package_overrides_for_builtin_materializations | 2024.04.141      | 2024.05.xxx         | 1.6.14, 1.7.14    | 1.8.0             |
+| require_resource_names_without_spaces                           | 2024.05.xxx      | 2024.06.xxx         | 1.8.0           | 1.9.0             |
```

Comment:
apologies but what does the xx mean?

Reply:
It means, I don't know what the exact release # is going to be, but the release is planned for `2024.05` (May) or `2024.06` (June). Definitely open to your thoughts — maybe `2024.05.TBD`, or just `2024.05` for now?

https://docs.getdbt.com/reference/commands/version#versioning

<img width="834" alt="Screenshot 2024-05-01 at 21 38 14" src="https://github.com/dbt-labs/docs.getdbt.com/assets/13897643/01ac126b-1f73-40c1-8fac-1e0096be45fc">

Reply:
Changed to TBD